server/liveblog/advertisements/advertisements.py

```diff
@@ -60,3 +67,33 @@ def on_updated(self, updates, original):
         # updates the collections, (caching issues)
         else:
             collections_service.update_advertisement(original)
+
+
+@advertisements_blueprint.route('/api/advertisements/<blog_id>/<output>/')
+def get_advertisements(blog_id, output):
+    """@TODO: add docstrings"""
+
+    # if the `output` is the `_id` get the data.
+    if output:
+        if isinstance(output, str):
+            output = get_resource_service('outputs').find_one(req=None, _id=output)
+        if not output:
+            return api_error('output not found', 404)
+        else:
+            collection = get_resource_service('collections').find_one(req=None, _id=output.get('collection'))
+            output['collection'] = collection
+
+    if output and output.get('collection', False):
+        ads_ids = output['collection'].get('advertisements', [])
+        ads_ids = list(map(lambda x: x['advertisement_id'], ads_ids))
+
+        ads_query = get_resource_service('advertisements').find({"_id": {"$in": ads_ids}})
+        ads = list(ads_query)
```

Comment:
shouldn't ads be defined empty list outside of the if ?
so `return api_response(ads, 200)`  will not break if no ads defined.

Reply:
Yeah, you're right. It will break if `if output and output.get('collection', False)` evaluate to False. Fixed